File: 02_devApplication_IOS_avec_Objective_C/04_firstResponder/04_firstResponder/ViewController.m
//
//  ViewController.m
//  04_firstResponder
//
//  Created by etudiant on 4/19/17.
//  Copyright © 2017 etudiant. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *txt1;

@property (weak, nonatomic) IBOutlet UITextField *txt2;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}
- (IBAction)btnCacherClavierTxt1:(id)sender
{
    [self.txt1 resignFirstResponder];
}

- (IBAction)btnCacherClavierTxt2:(id)sender
{
    [self.txt2 resignFirstResponder];
}
- (IBAction)btnFocusSutTxt:(id)sender
{
    [self.txt2 becomeFirstResponder];
}

//donner la possibilité a l'utilisateur àde faire diparaitre le clavier visuel en touchant la partie libre de l'écran

//Pour cela je dois overrider la méthode touch
//touchesBegan et arreter l'edition en cours(ou qu'elle soit)
-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [self.view endEditing:YES ];
}
//Methode de traitement de l'evenement 'did end on exit'
//qui est leve quand on touch
- (IBAction)btnRoutouClavierToched:(UITextView *)controle
{
    [controle resignFirstResponder];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 04_devApplication_IOS_avec_Swift/03_FirstResponder/03_FirstResponder/ViewController.swift
//
//  ViewController.swift
//  03_FirstResponder
//
//  Created by etudiant on 5/11/17.
//  Copyright © 2017 etudiant. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var txt1: UITextField!
    @IBOutlet weak var txt2: UITextField!

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func btnCacherClavierTxt1(_ sender: Any) {
        txt1.resignFirstResponder()
    }
    
    @IBAction func btnCacherClavierTxt2(_ sender: Any) {
        txt2.resignFirstResponder()
    }
    
    
    @IBAction func btnFocusTx2Touch(_ sender: Any) {
        txt2.becomeFirstResponder()
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        //solution 1
        //txt1.resignFirstResponder()
        //txt2.resignFirstResponder()
        
        //solution 2 = terminer edition pour tous
        view.endEditing(true)
    }
    
    //cacher clavier lors du click sur le return du clavier
    @IBAction func btnReturnCacherTxt1(_ sender: Any) {
        txt1.resignFirstResponder()
    }
    
    //cacher clavier lors du click sur le return du clavier
    @IBAction func btnReturnCacherTxt2(_ sender: Any) {
        txt1.resignFirstResponder()
    }

}

